# How Does a Bike-Share Navigate Speedy Success?

**Chao-Ting Fan**

## Introduction:
This project is the case study for the requirement from Google Data Analytics Capstone course.
This is a exploratory analysis that involves a fictional bikeshare company.The analysis will follow the 6 phases of the Data Analysis process: Ask, Prepare, Process, Analyze, and Act (APPAA).<br>
該項目是 Google 數據分析 Capstone 課程要求的案例研究。
這是一項涉及一家虛構的共享單車公司的探索性分析。該分析將遵循數據分析流程的 6 個階段：詢問、準備、處理、分析和行動 (APPAA)。
## Sope of Work:

**Ask and Identify the business task**
* Producing a scope of work document
* Framing a problem statement
* Defining project goals and the measurements to define success by discussing the project with stakeholders and ascertaining expectations<br>

**詢問並確定業務任務**
* 制定工作範圍文件
* 構建問題陳述
* 通過與利益相關者討論項目並確定期望來定義項目目標和衡量成功的標準<br>

**Extract and prepare the data for exploration**
* Identifying where data resides and its structure
* Determine the credibility and integrity of the data that meet analysis standard
* Addressing licensing, privacy, security, and accessibility concerns
* Verifying data integrity
* Identifying how the data helps answer the questions we're pursuing
* Filtering and sorting the data<br>

**提取並準備探索數據**
* 識別數據所在位置及其結構
* 解決許可、隱私、安全和可訪問性問題
* 驗證數據完整性
* 確定數據如何幫助回答我們正在追求的問題
* 對數據進行過濾和排序

**Process**
* Check for dirty data - in particular for data that is outdated, duplicated, incomplete, inconsistent, and inaccurate
* Wrangle and clean the data until we can apply unbiased and representative data analytics
* Document cleaning processes and results<br>

**處理**
* 檢查原始數據 - 特別是過時、重複、不完整、不一致和不准確的數據
* 整理和清理數據，直到我們能夠應用公正且具有代表性的數據分析
* 記錄清潔過程和結果

**Descriptive analysis**
* Ordanize and format the data
* Identify patterns(Using aggregations, calculations, visuals)
* Draw conclusions<br>

**描述性分析**
* 整理和格式化數據
* 識別模式（使用聚合、計算、視覺效果）
* 得出結論

**Share key finding**
* Create an effective and accessible presentation that answers the original business problem with compelling storytelling
* Share insights to stakeholders<br>

**分享關鍵發現**
* 創建有效且易於理解的演示文稿，通​​過引人入勝的故事講述來回答最初的業務問題
* 與利益相關者分享見解

**Act**
* Based on the analysis conducted, provide recommendations for the marketing campaign to move forward with
* Solve problems
<br>

**行動**
* 根據進行的分析，為營銷活動的推進提供建議
* 解決問題

## About the company
Cyclistic is a bike-share offering company established in 2016. Since then, the program has grown to a fleet of 5,824 bicycles that are geotracked and locked into a network of 692 stations across Chicago. The bikes can be unlocked from one station and returned to any other station in the system anytime. 
Cyclistic offers three pricing plan: Single-Ride Pass, Full-Day Pass, Annual Membership. Riders who purchase Single-Ride and Full-Day pass are refferred to as casual riders. Riders who purchase annual memberships are members.<br>

Cyclistic 是一家成立於 2016 年的自行車共享服務公司。從那時起，該計劃已發展到擁有 5,824 輛自行車的車隊，這些自行車經過地理跟踪並鎖定在芝加哥 692 個站點的網絡中。 自行車可以從一個站點解鎖並隨時返回到系統中的任何其他站點。
Cyclistic 提供三種定價計劃：單程通行證、全日通行證、年度會員資格。 購買單程和全日通行證的乘客被稱為休閒乘客。 購買年度會員資格的騎手就是會員。



<font size="5">1. Ask and Identify the Business Task </font>


**Case Backgroud:**

The director of Cyclistic thoerized maximizing the number of member will be the key to company’s success. Therefore, the marketing analyst needs to have a comprehensive understanding on how annual members and casual riders differ. After the executive approval, I’ll be designing a new marketing strategy and finding the data insights of turning casual riders into members.<br>

Cyclistic董事認為最大化會員數量將是公司成功的關鍵。 因此，營銷分析師需要全面了解年度會員和一般使用者的區別。 獲得高管批准後，我將設計新的營銷策略，並尋找將一般使用者轉變為會員的數據見解。

**Discussing Goals and Expectations**

| Stakeholders | Expectations | Projects/Business Goals |
|---|---|---|
|Cyclistic Executive Team| Compelling, relevant, and straightforward insights to inform data-driven marketing decisions | Implementing strategic initiatives to promote business growth 
|Lily Moreno, Director of Marketing| Evidence to back up her theory and marketing recommendations | Convert a sizable group of casual riders to become annual members | 
|Marketing analytics team| Uncovering the differences and motivations behind different customer types | Produce data-driven and actionable results to inform business decision making 

**Question:**

* How do annual members and casual riders use Cyclistic bikes differently?<br>
* How might we influence casual riders to purchase annual subscriptions based on their riding habits?<br>

* 年度會員和一般使用者使用自行車的方式有何不同？<br>
* 我們如何影響一般使用者根據他們的騎行習慣購買年度訂閱？

**Bussiness Task:**

* Finding and understanding how casual user and members use the sevice differently.
* Provide effective visuals and relevant data to support insights
* Analyze rider's usage patterns to give recommendations for marketing membership conversion program.<br>

* 發現並了解一般使用者和會員如何以不同的方式使用服務。
* 提供有效的視覺效果和相關數據來支持見解
* 分析騎乘者的使用模式，為營銷會員轉化計劃提供建議。

<font size="5">2. Prepare and Organize the Relevant Data</font>

**Task:**

 1. Identify how the data is organized
 2. Determine the credibility of the data.
 3. Sort and Filter the data as necessary
 4. Download and store the data <br>

**任務：**

  1. 確定數據的組織方式
  2. 確定數據的可信度。
  3. 根據需要對數據進行排序和過濾
  4. 下載並存儲數據
#### Data Sources 

A total of **12 datasets** have been made available for each month starting from **May 2021 to April 2022**.  The data used in this case study is sourced from Motivate International Inc under [this agreement][1] and were obtained from [Divvy Trip Data][2]. Each data point represents a single bike trip from one docking station to the next and it captures the details of every ride logged by the customers of Cyclistic. This data that has been made publicly available has been scrubbed to omit rider's personal information.<br>

從 **2021 年 5 月到 2022 年 4 月**，每個月總共提供 **12 個數據集**。 本案例研究中使用的數據源自 [本協議][1] 下的 Motivate International Inc，並從 [Divvy Trip Data][2] 獲得。 每個數據點代表從一個停靠站到下一個停靠站的單次自行車行程，它捕獲 Cyclistic 客戶記錄的每次騎行的詳細信息。 這些已公開的數據已被刪除，以屏蔽用戶的個人信息。

[1]: https://ride.divvybikes.com/data-license-agreement
[2]: https://divvy-tripdata.s3.amazonaws.com/index.html

In [1]:
# Load packages
library(tidyverse)
library(lubridate)
library(geosphere)
library(skimr)
library(data.table)

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()


Attaching package: ‘lubridate’


The following objects are masked from ‘package:base’:

    date, intersect, setdiff, union



Attaching package: ‘data.table’


The following objects are masked from ‘package:lubridate’:

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year


The following objects are masked from ‘package:dplyr’:

    between, first, last


The following object is masked from ‘package:purrr’:

    transpose




In [2]:
# Importing datasets
may_21 <- read_csv("../input/cyclistic-capstone/202105-divvy-tripdata.csv")
jun_21 <- read_csv("../input/cyclistic-capstone/202106-divvy-tripdata.csv")
jul_21 <- read_csv("../input/cyclistic-capstone/202107-divvy-tripdata.csv")
aug_21 <- read_csv("../input/cyclistic-capstone/202108-divvy-tripdata.csv")
sep_21 <- read_csv("../input/cyclistic-capstone/202109-divvy-tripdata.csv")
oct_21 <- read_csv("../input/cyclistic-capstone/202110-divvy-tripdata.csv")
nov_21 <- read_csv("../input/cyclistic-capstone/202111-divvy-tripdata.csv")
dec_21 <- read_csv("../input/cyclistic-capstone/202112-divvy-tripdata.csv")
jan_22 <- read_csv("../input/cyclistic-capstone/202201-divvy-tripdata.csv")
feb_22 <- read_csv("../input/cyclistic-capstone/202202-divvy-tripdata.csv")
mar_22 <- read_csv("../input/cyclistic-capstone/202203-divvy-tripdata.csv")
apr_22 <- read_csv("../input/cyclistic-capstone/202204-divvy-tripdata.csv")

Rows: 531633 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (7): ride_id, rideable_type, start_station_name, start_station_id, end_...
dbl  (4): start_lat, start_lng, end_lat, end_lng
dttm (2): started_at, ended_at

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 729595 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (7): ride_id, rideable_type, start_station_name, start_station_id, end_...
dbl  (4): start_lat, start_lng, end_lat, end_lng
dttm (2): started_at, ended_at

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 822410 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (7

In [3]:
# Merging the datasets
all_trips <- rbind(may_21, jun_21, jul_21, aug_21, sep_21, oct_21, nov_21,
                    dec_21, jan_22, feb_22, mar_22, apr_22)

head(all_trips)

ride_id          rideable_type started_at          ended_at           
1 C809ED75D6160B2A electric_bike 2021-05-30 11:58:15 2021-05-30 12:10:39
2 DD59FDCE0ACACAF3 electric_bike 2021-05-30 11:29:14 2021-05-30 12:14:09
3 0AB83CB88C43EFC2 electric_bike 2021-05-30 14:24:01 2021-05-30 14:25:13
4 7881AC6D39110C60 electric_bike 2021-05-30 14:25:51 2021-05-30 14:41:04
5 853FA701B4582BAF electric_bike 2021-05-30 18:15:39 2021-05-30 18:22:32
6 F5E63DFD96B2A737 electric_bike 2021-05-30 11:33:41 2021-05-30 11:57:17
  start_station_name start_station_id end_station_name end_station_id start_lat
1 NA                 NA               NA               NA             41.90    
2 NA                 NA               NA               NA             41.88    
3 NA                 NA               NA               NA             41.92    
4 NA                 NA               NA               NA             41.92    
5 NA                 NA               NA               NA             41.94    
6 NA                 NA               NA               NA             41.88    
  start_lng end_lat end_lng member_casual
1 -87.63    41.89   -87.61  casual       
2 -87.62    41.79   -87.58  casual       
3 -87.70    41.92   -87.70  casual       
4 -87.70    41.94   -87.69  casual       
5 -87.69    41.94   -87.70  casual       
6 -87.63    41.89   -87.62  casual

**1.  Null values were observed for start station name, start station ID, end station name and end station IDs, end_lat, end-lang.**

In [4]:
skim_without_charts(all_trips)

{"character":[{"skim_variable":"ride_id","n_missing":0,"complete_rate":1,"min":16,"max":16,"empty":0,"n_unique":5757551,"whitespace":0},{"skim_variable":"rideable_type","n_missing":0,"complete_rate":1,"min":11,"max":13,"empty":0,"n_unique":3,"whitespace":0},{"skim_variable":"start_station_name","n_missing":790207,"complete_rate":0.8628,"min":3,"max":53,"empty":0,"n_unique":865,"whitespace":0},{"skim_variable":"start_station_id","n_missing":790204,"complete_rate":0.8628,"min":3,"max":44,"empty":0,"n_unique":856,"whitespace":0},{"skim_variable":"end_station_name","n_missing":843361,"complete_rate":0.8535,"min":10,"max":53,"empty":0,"n_unique":864,"whitespace":0},{"skim_variable":"end_station_id","n_missing":843361,"complete_rate":0.8535,"min":3,"max":44,"empty":0,"n_unique":856,"whitespace":0},{"skim_variable":"member_casual","n_missing":0,"complete_rate":1,"min":6,"max":6,"empty":0,"n_unique":2,"whitespace":0}],"numeric":[{"skim_variable":"start_lat","n_missing":0,"complete_rate":1,"mean":41.9003,"sd":0.0468,"p0":41.64,"p25":41.881,"p50":41.8986,"p75":41.9289,"p100":45.635},{"skim_variable":"start_lng","n_missing":0,"complete_rate":1,"mean":-87.6465,"sd":0.0302,"p0":-87.84,"p25":-87.6603,"p50":-87.6427,"p75":-87.6278,"p100":-73.7965},{"skim_variable":"end_lat","n_missing":4766,"complete_rate":0.9992,"mean":41.9006,"sd":0.0469,"p0":41.39,"p25":41.8813,"p50":41.899,"p75":41.9291,"p100":42.1681},{"skim_variable":"end_lng","n_missing":4766,"complete_rate":0.9992,"mean":-87.6467,"sd":0.0298,"p0":-88.97,"p25":-87.661,"p50":-87.6429,"p75":-87.6278,"p100":-87.49}],"POSIXct":[{"skim_variable":"started_at","n_missing":0,"complete_rate":1,"min":"2021-05-01 00:00:11","max":"2022-04-30 23:59:54","median":"2021-08-31 17:17:20","n_unique":4821094},{"skim_variable":"ended_at","n_missing":0,"complete_rate":1,"min":"2021-05-01 00:03:26","max":"2022-05-02 00:35:01","median":"2021-08-31 17:34:09","n_unique":4815847}]}

── Data Summary ────────────────────────
                           Values   
Name                       all_trips
Number of rows             5757551  
Number of columns          13       
_______________________             
Column type frequency:              
  character                7        
  numeric                  4        
  POSIXct                  2        
________________________            
Group variables            None     

── Variable type: character ────────────────────────────────────────────────────
  skim_variable      n_missing complete_rate min max empty n_unique whitespace
1 ride_id                    0         1      16  16     0  5757551          0
2 rideable_type              0         1      11  13     0        3          0
3 start_station_name    790207         0.863   3  53     0      865          0
4 start_station_id      790204         0.863   3  44     0      856          0
5 end_station_name      843361         0.854  10  53     0      864        

In [5]:
# remove null values
all_trips <- drop_na(all_trips)
# skim_without_charts(all_trips)

**2. Rename columns for better readability** 

In [6]:
all_trips <- all_trips %>%
    rename(ride_type = rideable_type, 
           start_time = started_at,
           end_time = ended_at,
           customer_type = member_casual)
glimpse(all_trips)

Rows: 4,615,748
Columns: 13
$ ride_id            <chr> "0F3AE375DEC608D9", "43E916C72D4C77B3", "2AAB47362E…
$ ride_type          <chr> "classic_bike", "electric_bike", "electric_bike", "…
$ start_time         <dttm> 2021-05-02 08:21:19, 2021-05-01 22:15:55, 2021-05-…
$ end_time           <dttm> 2021-05-02 09:20:46, 2021-05-01 22:51:27, 2021-05-…
$ start_station_name <chr> "Michigan Ave & Oak St", "Michigan Ave & Oak St", "…
$ start_station_id   <chr> "13042", "13042", "TA1307000127", "604", "13042", "…
$ end_station_name   <chr> "Michigan Ave & Oak St", "Michigan Ave & Oak St", "…
$ end_station_id     <chr> "13042", "13042", "TA1309000014", "604", "13042", "…
$ start_lat          <dbl> 41.90096, 41.90104, 41.95093, 42.05824, 41.90100, 4…
$ start_lng          <dbl> -87.62378, -87.62379, -87.65915, -87.67743, -87.623…
$ end_lat            <dbl> 41.90096, 41.90078, 41.96709, 42.05824, 41.90099, 4…
$ end_lng            <dbl> -87.62378, -87.62385, -87.66739, -87.67743, -87.623…
$ customer_t

<font size="5">3. Process: - Clean the Data </font>


### Clean and prepare the data for analysis



**1. Add new columns that can be used for aggregate functions**

In [7]:
#column for month when the trip started
all_trips$month <- format(as.Date(all_trips$start_time),'%b')

#column for day of the week the trip started
all_trips$day_of_the_week <- format(as.Date(all_trips$start_time),'%a')

# Creating hour column
all_trips <- all_trips %>%
    mutate(hour = strftime(all_trips$end_time, "%H"))

**calculate some important informations and columns for analysis**

**2. Calculating Trip Duration and Ride Distance**

In [8]:
all_trips <- mutate(all_trips,trip_duration = difftime(all_trips$end_time,all_trips$start_time, units = "mins"))
head(all_trips)

all_trips$ride_distance <- distGeo(matrix(c(all_trips$start_lng, all_trips$start_lat), ncol = 2),
                              matrix(c(all_trips$end_lng, all_trips$end_lat), ncol = 2))

all_trips$ride_distance <- all_trips$ride_distance/1000

ride_id          ride_type     start_time          end_time           
1 0F3AE375DEC608D9 classic_bike  2021-05-02 08:21:19 2021-05-02 09:20:46
2 43E916C72D4C77B3 electric_bike 2021-05-01 22:15:55 2021-05-01 22:51:27
3 2AAB47362E76AC7F electric_bike 2021-05-31 00:50:15 2021-05-31 00:56:59
4 344011882FBBB8BD classic_bike  2021-05-31 07:35:08 2021-05-31 08:33:45
5 238E58CAB07FA2A8 electric_bike 2021-05-02 16:55:59 2021-05-02 16:57:10
6 BF99592737266B31 classic_bike  2021-05-07 06:03:21 2021-05-07 07:12:20
  start_station_name          start_station_id end_station_name           
1 Michigan Ave & Oak St       13042            Michigan Ave & Oak St      
2 Michigan Ave & Oak St       13042            Michigan Ave & Oak St      
3 Clark St & Grace St         TA1307000127     Clark St & Leland Ave      
4 Sheridan Rd & Noyes St (NU) 604              Sheridan Rd & Noyes St (NU)
5 Michigan Ave & Oak St       13042            Michigan Ave & Oak St      
6 Michigan Ave & Oak St       13042            Michigan Ave & Oak St      
  end_station_id start_lat start_lng end_lat  end_lng   customer_type month
1 13042          41.90096  -87.62378 41.90096 -87.62378 casual        May  
2 13042          41.90104  -87.62379 41.90078 -87.62385 casual        May  
3 TA1309000014   41.95093  -87.65915 41.96709 -87.66739 casual        May  
4 604            42.05824  -87.67743 42.05824 -87.67743 casual        May  
5 13042          41.90100  -87.62374 41.90099 -87.62373 casual        May  
6 13042          41.90096  -87.62378 41.90096 -87.62378 casual        May  
  day_of_the_week hour trip_duration 
1 Sun             09   59.450000 mins
2 Sat             22   35.533333 mins
3 Mon             00    6.733333 mins
4 Mon             08   58.616667 mins
5 Sun             16    1.183333 mins
6 Fri             07   68.983333 mins

**4. Check if there is negative trip duration and ride distance.
     There are also some trips where bikes were taken out for checking and put back into the station and took a time period of less than a minute. If there is data like this, 
we need to remove that**

In [9]:
min(all_trips$trip_duration)

Time difference of -55.9 mins

As we can see the minimum time is -55.9 mins shows that there is negative data present so we need to get rid of that.

In [10]:
all_trips <- subset(all_trips,trip_duration > 0)
min(all_trips$trip_duration)

Time difference of 0.01666667 mins

In [11]:
all_trips <- subset(all_trips,!(start_station_name == end_station_name & trip_duration <= 1))
all_trips <- subset(all_trips,trip_duration >= 1)
min(all_trips$trip_duration)

Time difference of 1 mins

**5. remove the trips that were part of quality tests by the company**

In [12]:
#checking for testrides that were made by company for quality checks
nrow(subset(all_trips, start_station_name %like% "TEST"))
nrow(subset(all_trips, start_station_name %like% "test"))
nrow(subset(all_trips, start_station_name %like% "Test"))

[1] 0

[1] 0

[1] 1

In [13]:
#remove test rides
all_trips<- all_trips[!(all_trips$start_station_name %like% "test"),]

**6.The geographic data show the precise slot in each station of where users have take and return their bike, however, in order to show the usage of bike stations in the visualization, I unify users locations in the same station with median value**

In [14]:
start_station_loc <- all_trips %>% 
    group_by(start_station_name) %>%
    summarise(start_lat_n = median(start_lat),start_lng_n = median(start_lng))

end_station_loc <- all_trips %>% 
    group_by(end_station_name) %>%
    summarise(end_lat_n = median(end_lat),end_lng_n = median(end_lng))

In [15]:
df <- merge(x = all_trips, y = start_station_loc,by='start_station_name')
all_trips <- merge(x = df, y = end_station_loc,by='end_station_name')

**7. make sure that customer_type column has only two distinct values.**

In [16]:
# checking count of distinct values
table(all_trips$customer_type)
#aggregating total trip duration by customer type
setNames(aggregate(trip_duration ~ customer_type, all_trips, sum), c("customer_type", "total_trip_duration(mins)"))


 casual  member 
1994857 2558720 

customer_type total_trip_duration(mins)
1 casual        63262817 mins            
2 member        33153613 mins

**8. Remove unnecessary columns.**

In [17]:
all_trips <- select(all_trips,-start_time,-end_time,-start_station_id,-end_station_id, -start_lat,-start_lng,-end_lat,-end_lng)
head(all_trips)

end_station_name    start_station_name       ride_id          ride_type    
1 2112 W Peterson Ave Sheridan Rd & Loyola Ave 5CAC8BF9A2444E8F classic_bike 
2 2112 W Peterson Ave Clark St & Elmdale Ave   DC331605A1D50790 classic_bike 
3 2112 W Peterson Ave Warren Park East         98D322B9442433C4 classic_bike 
4 2112 W Peterson Ave Clark St & Bryn Mawr Ave 518FDEAAFA9B460F classic_bike 
5 2112 W Peterson Ave Clark St & Schreiber Ave 62D6A3C01D203683 classic_bike 
6 2112 W Peterson Ave Damen Ave & Foster Ave   F8FD6C59F1E05B93 electric_bike
  customer_type month day_of_the_week hour trip_duration  ride_distance
1 casual        Oct   Fri             17   23.850000 mins 2.154958     
2 member        Nov   Mon             01    7.016667 mins 1.149791     
3 member        Mar   Sun             05   19.016667 mins 1.504896     
4 member        May   Tue             23    9.200000 mins 1.463367     
5 member        Oct   Fri             20    8.466667 mins 1.352286     
6 member        Nov   Sat             13    8.983333 mins 1.753117     
  start_lat_n start_lng_n end_lat_n end_lng_n
1 42.00104    -87.66120   41.99118  -87.68359
2 41.99086    -87.66972   41.99118  -87.68359
3 42.00455    -87.68067   41.99118  -87.68359
4 41.98359    -87.66915   41.99118  -87.68359
5 41.99925    -87.67138   41.99118  -87.68359
6 41.97561    -87.67946   41.99118  -87.68359

**9. Reorder the columns**

In [18]:
all_trips <- all_trips[, c(3, 4, 5, 2, 1, 11, 12, 13, 14, 6, 7, 8, 9, 10 )]

**10. Remove the data that visited the place less than 500 times. This step let us focus on significant differences between user's number of visits on each station without outliers.**

In [19]:
data<- all_trips %>% group_by(end_station_name) %>% 
                    summarise(number_of_rides = n()) %>% arrange(desc(number_of_rides))
data <- filter(data,number_of_rides<500)
station_d <- data$end_station_name
all_trips <- filter(all_trips,!end_station_name %in% station_d)

In [20]:
head(all_trips)

ride_id          ride_type     customer_type start_station_name      
1 5CAC8BF9A2444E8F classic_bike  casual        Sheridan Rd & Loyola Ave
2 DC331605A1D50790 classic_bike  member        Clark St & Elmdale Ave  
3 98D322B9442433C4 classic_bike  member        Warren Park East        
4 518FDEAAFA9B460F classic_bike  member        Clark St & Bryn Mawr Ave
5 62D6A3C01D203683 classic_bike  member        Clark St & Schreiber Ave
6 F8FD6C59F1E05B93 electric_bike member        Damen Ave & Foster Ave  
  end_station_name    start_lat_n start_lng_n end_lat_n end_lng_n month
1 2112 W Peterson Ave 42.00104    -87.66120   41.99118  -87.68359 Oct  
2 2112 W Peterson Ave 41.99086    -87.66972   41.99118  -87.68359 Nov  
3 2112 W Peterson Ave 42.00455    -87.68067   41.99118  -87.68359 Mar  
4 2112 W Peterson Ave 41.98359    -87.66915   41.99118  -87.68359 May  
5 2112 W Peterson Ave 41.99925    -87.67138   41.99118  -87.68359 Oct  
6 2112 W Peterson Ave 41.97561    -87.67946   41.99118  -87.68359 Nov  
  day_of_the_week hour trip_duration  ride_distance
1 Fri             17   23.850000 mins 2.154958     
2 Mon             01    7.016667 mins 1.149791     
3 Sun             05   19.016667 mins 1.504896     
4 Tue             23    9.200000 mins 1.463367     
5 Fri             20    8.466667 mins 1.352286     
6 Sat             13    8.983333 mins 1.753117

**11.   export the data.**

In [21]:
# Write csv of trips
write.table(all_trips, file='./tripsdata.csv',sep=",",row.names=F, na = "NA")

**Now that our data is clean and calculated time to move to next phase.**


<font size="5">4. Analyse: - Identify Trends and Relationships</font>

<font size="3">The dataframe is now ready for descriptive analysis that will help us uncover some insights on how the casual riders and members use Cyclistic rideshare differently.

In [22]:
# User count
usercount <- all_trips %>% 
    group_by(customer_type) %>% 
    summarise(count = length(ride_id),
              'percentage' = (length(ride_id) / nrow(all_trips)) * 100)

usercount$percentage <- round(usercount$percentage, digits = 0)

usercount

customer_type count   percentage
1 casual        1968160 44        
2 member        2545582 56

In [23]:
#Ride distance by customer type
ride_distance_data <- all_trips %>%
    group_by(customer_type) %>%
    summarise('distance'=mean(ride_distance),median(ride_distance))

ride_distance_data

customer_type distance median(ride_distance)
1 casual        2.211575 1.720129             
2 member        2.080069 1.549695

Both users don't seem to have significant difference in ride distance

In [24]:
# Bike use count by user
bikeused <- all_trips %>% 
    group_by(customer_type, ride_type) %>%
    summarise(count = length(ride_type),
         'percentage' = length(ride_type) / nrow(all_trips) * 100,
         averagetime = mean(trip_duration),.groups = 'drop')

bikeused$percentage <- round(bikeused$percentage, digits = 0)
bikeused$averagetime <- round(bikeused$averagetime, digits = 0)

bikeused

customer_type ride_type     count   percentage averagetime
1 casual        classic_bike  1204159 27         26 mins    
2 casual        docked_bike    285048  6         68 mins    
3 casual        electric_bike  478953 11         20 mins    
4 member        classic_bike  1929742 43         13 mins    
5 member        electric_bike  615840 14         12 mins

casual riders tend to use the service longer than members for every kind of bike type 

**1. Get some simple statistics on trip_duration by customer_type.**

In [25]:
#statistical summary of trip_duration by customer_type
all_trips %>%
    group_by(customer_type) %>%
    summarise(min_trip_duration = min(trip_duration),max_trip_duration = max(trip_duration),
              median_trip_duration = median(trip_duration), mean_trip_duration = mean(trip_duration))

customer_type min_trip_duration max_trip_duration median_trip_duration
1 casual        1 mins            55944.150 mins    16.400000 mins      
2 member        1 mins             1495.633 mins     9.466667 mins      
  mean_trip_duration
1 30.32883 mins     
2 12.92422 mins

The mean trip duration of member riders is lower than the mean trip duration of all trips, while it is exactly the opposite for casual riders, whose mean trip duration is higher than the the mean trip duration of all trips. This tells us that **casual riders usually take the bikes out for a longer duration compared to members.**

**2. Calculating key info in customer types based on Months, Weekday**

In [26]:
#Calculating key info based on Months
all_trips$month <- ordered(all_trips$month,
                                        levels=c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"))
user_type_monthly_data <- all_trips %>% 
  group_by(month,customer_type) %>% 
  summarise(number_of_rides = n(),
            total_ride_time = sum(trip_duration),
            average_ride_time = mean(trip_duration))

user_type_monthly_data

`summarise()` has grouped output by 'month'. You can override using the
`.groups` argument.


month customer_type number_of_rides total_ride_time average_ride_time
1  Jan   casual         12194            298733.1 mins 24.49837 mins    
2  Jan   member         66236            687895.2 mins 10.38552 mins    
3  Feb   casual         14646            364000.9 mins 24.85326 mins    
4  Feb   member         72349            781287.9 mins 10.79888 mins    
5  Mar   casual         64913           1777674.9 mins 27.38550 mins    
6  Mar   member        145317           1736213.6 mins 11.94777 mins    
7  Apr   casual         88496           2292139.3 mins 25.90105 mins    
8  Apr   member        175583           2064400.6 mins 11.75741 mins    
9  May   casual        211937           7957591.9 mins 37.54697 mins    
10 May   member        229510           3336564.3 mins 14.53777 mins    
11 Jun   casual        297293          10582813.2 mins 35.59725 mins    
12 Jun   member        298636           4281008.3 mins 14.33521 mins    
13 Jul   casual        361548          11715100.1 mins 32.40261 mins    
14 Jul   member        316485           4426641.5 mins 13.98689 mins    
15 Aug   casual        334188           9475994.2 mins 28.35528 mins    
16 Aug   member        326592           4479931.0 mins 13.71721 mins    
17 Sep   casual        286326           7919751.9 mins 27.65991 mins    
18 Sep   member        321677           4270020.7 mins 13.27425 mins    
19 Oct   casual        184851           4788971.7 mins 25.90720 mins    
20 Oct   member        282847           3442914.8 mins 12.17236 mins    
21 Nov   casual         68109           1517858.5 mins 22.28573 mins    
22 Nov   member        181865           2016575.8 mins 11.08831 mins    
23 Dec   casual         43659           1001361.5 mins 22.93597 mins    
24 Dec   member        128485           1376210.8 mins 10.71106 mins

We can see that as we approach the warmer months the number of rides tends to increase and as we move towards colder months the total number of rides starts to fall.

In [27]:
#Calculating key info based on Weekday
all_trips$day_of_the_week <- ordered(all_trips$day_of_the_week,
                                        levels=c("Mon","Tue", "Wed", "Thu", "Fri", "Sat", "Sun" ))
user_type_weekday_data <- all_trips %>% 
  group_by(day_of_the_week, customer_type) %>% 
  summarise(number_of_rides = n(),
            total_ride_time = sum(trip_duration),
            average_ride_time = mean(trip_duration))

user_type_weekday_data

`summarise()` has grouped output by 'day_of_the_week'. You can override using
the `.groups` argument.


day_of_the_week customer_type number_of_rides total_ride_time
1  Mon             casual        220152           6739212 mins  
2  Mon             member        353745           4404222 mins  
3  Tue             casual        200822           5204392 mins  
4  Tue             member        397492           4759043 mins  
5  Wed             casual        212985           5554413 mins  
6  Wed             member        404907           4969190 mins  
7  Thu             casual        223829           5996506 mins  
8  Thu             member        384631           4697349 mins  
9  Fri             casual        273471           7675395 mins  
10 Fri             member        353286           4451161 mins  
11 Sat             casual        451912          14848466 mins  
12 Sat             member        346930           5076728 mins  
13 Sun             casual        384989          13673607 mins  
14 Sun             member        304591           4541971 mins  
   average_ride_time
1  30.61163 mins    
2  12.45027 mins    
3  25.91545 mins    
4  11.97268 mins    
5  26.07889 mins    
6  12.27242 mins    
7  26.79057 mins    
8  12.21261 mins    
9  28.06658 mins    
10 12.59931 mins    
11 32.85699 mins    
12 14.63329 mins    
13 35.51688 mins    
14 14.91170 mins

Members have a steady riding throughout the days of the week but casual riders take most number of rides during the weekends.

**3. Check top 20 locations that each customer type visit**

In [28]:
casual_stantion_count <- all_trips %>% group_by(end_station_name,customer_type) %>% 
                    summarise(number_of_rides = n()) %>% arrange(desc(number_of_rides)) %>% filter(customer_type=="casual")
                        
head(casual_stantion_count,20)

member_stantion_count <- all_trips %>% group_by(end_station_name,customer_type) %>% 
                    summarise(number_of_rides = n()) %>% arrange(desc(number_of_rides)) %>% filter(customer_type=="member")
                        
head(member_stantion_count,20)

`summarise()` has grouped output by 'end_station_name'. You can override using
the `.groups` argument.


end_station_name                   customer_type number_of_rides
1  Streeter Dr & Grand Ave            casual        68509          
2  Millennium Park                    casual        32321          
3  Michigan Ave & Oak St              casual        29153          
4  Theater on the Lake                casual        21149          
5  Shedd Aquarium                     casual        20729          
6  Wells St & Concord Ln              casual        18110          
7  DuSable Lake Shore Dr & North Blvd casual        18086          
8  DuSable Lake Shore Dr & Monroe St  casual        17427          
9  Wabash Ave & Grand Ave             casual        16045          
10 Clark St & Lincoln Ave             casual        15852          
11 Clark St & Armitage Ave            casual        15179          
12 Lake Shore Dr & North Blvd         casual        14808          
13 Indiana Ave & Roosevelt Rd         casual        14737          
14 Wells St & Elm St                  casual        14561          
15 Clark St & Elm St                  casual        14413          
16 New St & Illinois St               casual        13465          
17 Lake Shore Dr & Monroe St          casual        13105          
18 Michigan Ave & Lake St             casual        13028          
19 Dusable Harbor                     casual        12935          
20 Fairbanks Ct & Grand Ave           casual        12855

`summarise()` has grouped output by 'end_station_name'. You can override using
the `.groups` argument.


end_station_name             customer_type number_of_rides
1  Kingsbury St & Kinzie St     member        24248          
2  Clark St & Elm St            member        23414          
3  Wells St & Concord Ln        member        23204          
4  Wells St & Elm St            member        20490          
5  Dearborn St & Erie St        member        18673          
6  Clinton St & Madison St      member        18299          
7  Wells St & Huron St          member        17856          
8  St. Clair St & Erie St       member        17571          
9  Clinton St & Washington Blvd member        17458          
10 Broadway & Barry Ave         member        17213          
11 Green St & Madison St        member        16695          
12 Wabash Ave & Grand Ave       member        15478          
13 Clark St & Armitage Ave      member        15201          
14 Wells St & Hubbard St        member        15023          
15 Clark St & Lincoln Ave       member        14881          
16 Wilton Ave & Belmont Ave     member        14672          
17 Desplaines St & Kinzie St    member        14645          
18 University Ave & 57th St     member        14626          
19 Ellis Ave & 60th St          member        14603          
20 Dearborn Pkwy & Delaware Pl  member        14102

From these result, we can see most of the locations visit by casual users are likely to be the tourist spots. On the contrary, members usually visit financial district where offices are located.

<font size="5">5. Share key finding</font>

The descriptive analysis and with support of the visualization on tableau, helps us uncover key findings on the pattern of how users use the sevice differently, here are some insights.<br>

描述性分析和 Tableau 可視化的支持，幫助我們發現有關用戶如何以不同方式使用服務的模式的關鍵發現，以下是一些見解。
The vistual analysis on tabeau click [here][1].

<font size="3">**Both User Type**</font>

The warmer months the number of rides tends to increase and as we move towards colder months the total number of rides starts to fall.<br>

溫暖的月份，乘車次數往往會增加，而當我們進入寒冷的月份時，乘車總數開始下降。

<font size="3">**Difference between users**</font>

|<font size="3">**Casuals**</font>|<font size="3">**Members**</font>|
|---|---|
| <font size="2">Despite Cylistics 所有用戶裡 56% of all users are member, Casual Riders take more number of rides during the **summer season** that is June, July, August.| <font size="2">Take more number of rides throughout the year except summer|
|<font size="2">Take most number of rides during the **weekends**.|<font size="2">Have a steady riding throughout the days of the week|
|<font size="2">The average using time is **more than twice longer** than members, even more longer in summer month|<font size="2">Tend to use the service in a short period of time|
|<font size="2">Most of the Rush is during the ***Afternoon Period*** |<font size="2">The annual member display two peak periods which is at **7 a.m and 5 p.m**|
|<font size="2">the top 20 stations visited by casual mostly are **tourist locations**. |<font size="2">usually visit financial district where offices are located|<br>

<font size="3">**用戶之間的差異**</font>

|<font size="3">**一般用戶**</font>|<font size="3">**會員**</font>|
|---|---|
| <font size="2">儘管 56% 的用戶是會員，但一般用戶在**夏季**（六月、七月、八月）乘坐的次數更多。| <font size="2">除夏季外，全年騎乘更多次數|
|<font size="2">在**週末**進行最多的騎乘次數。|<font size="2">一周中的每一天都保持穩定的騎乘次數|
|<font size="2">平均使用時間比會員**長兩倍以上**，夏季甚至更長|<font size="2">傾向於在短時間內使用該服務|
|<font size="2">大部分高峰期在***下午時段*** |<font size="2">年度會員有兩個高峰時段，即**上午 7 點和下午 5 點* *|
|<font size="2">一般使用者最常訪問的前20個車站大多是**旅遊地點**。 |<font size="2">經常訪問辦公室所在的金融區|<br>

### <font size="4">**Summary:**</font>

Our analysis reveals some obvious behavirol differences between two users that we can make a highly possible speculation: <br>
此分析揭示了兩個用戶之間一些明顯的行為差異，我們可以做出很有可能的推測：
#### 1. Casuals mainly use Cyclistic bikes for leisure activities
- Twice of usage during weekend compare to weekday
- Spend the majority of their bike trips near waterfront and leisure places.
- Spend significantly longer on average on every bike trip, suggesting that they spend time in-between docking stations doing leisurely activities 
- Do not use Cyclistic bikes often enough to buy an annual membership
#### 1.一般用戶主要使用自行車進行休閒活動
- 週末的使用量是工作日的兩倍
- 大部分自行車旅行都在海濱和休閒場所附近進行。
- 每次自行車旅行的平均花費時間明顯更長，這表明他們花時間在停靠站之間進行休閒活動
- 不要經常使用自行車來購買年度會員資格
#### 2. Members get more out of Cyclistic bikes by using them for leisure and commuting consistently
- Rely on bikes consistently each week and year-round, with no notable preference on a single day of the week
- Use Cyclistic bikes often during the rush hours on a typical workday
- Have a large geographical spread in the downtown area, particularly in high urban dense areas
- Preference for annual membership pass due to the often usage
#### 2. 會員持續使用自行車進行休閒和通勤，從而從自行車中獲得更多收益
- 每周和全年始終堅持騎自行車，一周中的某一天沒有明顯的偏好
- 在典型工作日的高峰時段經常使用自行車
- 在市中心地區擁有廣闊的地理分佈，特別是在城市高密度地區
- 由於經常使用，優先考慮年度會員通行證
    
[1]:https://public.tableau.com/shared/MTT7B5KQQ?:display_count=n&:origin=viz_share_link

<font size="5">6. Act</font>

Key Objective: Provide actionable insights and recommendations to the stakeholders that answer the business question.<br>
主要目標：為回答業務問題的利益相關者提供可行的見解和建議。
### (A.)**Actionable Plan:**

**Focus on the conversion of user**

1. Inform the casual riders how much they can save if they were to convert to a member by providing promotions throughout the year such as a free trial or a discount throughout the summer months.
<br>告知一般用戶，如果他們通過全年提供促銷活動（例如免費試用或整個夏季的折扣）轉為會員可以節省多少錢。
2. Have a advertising campaign to casual riders in the leading summer months and also focus on the time around 7am to 6pm.
<br>在夏季的幾個月中針對一般用戶開展廣告活動，並重點關注上午 7 點至下午 6 點左右的時間。
3. Provide attractive promotions for casual riders on weekdays / Offering weekend pass promotion for convering to Membership
<br>平日為一般用戶提供有吸引力的促銷/提供週末通行證促銷以轉換為會員
4. Offer discounted pricing during non-busy hours<br>
在非工作時段提供折扣

5. Since casual riders take more time on the service. It could be beneficial to promote more economical 'longer duration ride' for Membership riders
<br>由於一般用戶需要更多的服務時間。 為會員提供更經濟的“較長時間騎行”可能是有益的
6. Build parterships with famous attractions near the bay area with rewarding campaign.
<br>與灣區周邊著名景點建立合作夥伴關係，開展獎勵活動。

**Consider alternatives to conversion, such as new service, pricing models, campaign**

1. We can examine any options to innovate and enhance the current Cyclistic experience, with this primiliary analysis, we noticed that the user's actual needs can affects choice of the service plan. In particular, a three-tiered pricing approach could optimize revenue by offering a service that fits between single-use passes and an annual subscription that provides unlimited rides. This offering would seek a balance between the two, finding the sweet spot for casuals that cannot justify the economics of an expensive annual payment and the limitations of single-use passes. For example, a weekend pass would serve as a good compromise for users that typically only use Cyclistic for leisure on weekends.<br>
Cyclistic的未來不應局限於單一狹隘的戰略來實現可持續的業務增長。 我們應該研究任何創新和增強當前自行車體驗的選項，即使它是激進的。 特別是，三級定價方法可以通過提供介於一次性通行證和提供無限乘車的年度訂閱之間的服務來優化收入。 這項服務將尋求兩者之間的平衡，為那些無法證明昂貴的年費經濟性和一次性通行證的局限性的休閒人士找到最佳平衡點。 例如，週末通行證對於通常只在周末使用 Cyclistic 進行休閒的用戶來說是一個很好的折衷方案。<br>

6. As the bikes are likely used for leisure activities on the weekends, create an Ad campaigns that indicate the benifit of consistent using the service such as good health and sustainibility.
positive campaigns could also be made showing people using the bikes for exercise during the weeks. The ads could focus on how practical and consistent the bikes can be.<br>
由於自行車在週末可能會用於休閒活動，因此可以創建一系列廣告活動，強調持續使用該服務的好處，例如身體健康和可持續性。正面的廣告活動還可以展示人們在工作日使用自行車鍛煉，著重強調自行車的實用性和持續性。這些廣告可以突顯展示人們如何在週末利用自行車進行休閒活動，強調騎行對於身體健康和環保的積極影響。另外，也可以製作廣告，展示人們如何在工作日使用自行車進行鍛煉，強調自行車的實用性和持續性，使廣告更具吸引力。

#### B) **Reiterating on Data Extraction and Analysis**

To confirm that our speculations are true and uncover any other notable behavioral differences, we can consider some ways to understand more about our user<br>
為了確認我們的猜測是真實的並發現任何其他顯著的行為差異，我們可以考慮一些方法來更多地了解我們的用戶
1. Survey a significant sample of our user population to discover what truly defines each user group's behavior. Gravitating towards qualitative data that would provide insights like opinions and motivations (what do they use Cyclistic bikes for?) gives some much-needed context behind our initial findings. <br>
對我們用戶群的重要樣本進行調查，以發現真正定義每個用戶組行為的因素。 傾向於提供意見和動機等見解的定性數據（他們使用自行車的用途是什麼？）為我們的初步發現提供了一些急需的背景信息。<br>

2. Collecting more quantitative data into user demographics with features like income, age, and weight, we can built models to see more profound finding.<br>
收集更多的用戶人口統計數據，包括收入、年齡和體重等特徵，我們可以建立模型來看到更深刻的發現。




